proposals/interface-types/working-notes/Types.md

```diff
@@ -1,30 +1,55 @@
 # Interface value types
 
-This proposal defines new interface values, which have corresponding interface value types. Interface value types are a superset of the [core value types](https://webassembly.github.io/spec/core/syntax/types.html#value-types)
+The set of types and type constructors that are specific to Interface Types
+allow API designers to express types of data and function values in a way that
+is suitable for module-level descriptions. These types are considerably richer
+than those of core WebAssembly, but would not typically be enough for a full
+fledged programming language -- that is not the purpose.
 
-Some interface value types are compound by being constructed using one or more interface value types. The abstract syntax is designed to prevent recursive types by not allowing a compound type definition to refer to itself or other definitions. This is so that the set of control flow needed to generate all interface values is simple enough to allow for optimized lazy semantics.
+The space of types forms a a superset of the
+[core value types](https://webassembly.github.io/spec/core/syntax/types.html#value-types);
+this is because adapter functions will often use a mixture of core WebAssembly
+types and Interface types.
 
-The interface value type syntax is summarized below. Following after that is a section for each type that gives the full syntax, possible values, validation rules, and subtyping relationships.
+As a matter of policy, Interface types allow designers to express numeric types,
+strings, compound types, function types, optional types, arrays and
+sequences. Note that these types are intended to represent _immutable_ values;
+with some limited exceptions planned for the future.
+
+This document has a structure that mirrors the structure of the core WebAssembly
+proposal: we discuss the abstract structure of Interface types, the validation
+rules for type expressions and the binary format of type expressions.
+
+## Structure
+
+The Interface types can partitioned into different groups, reflecting the kind of entities involved.
+
+>Note: unlike core WebAssembly types, there is not always a straightforward
+>correspondence between Interface types and run-time entities. This is because
+>Interface types are _abstract_ and require a _realization_ in terms of core
+>WebAssembly entities.
+
+>This realization is not directly part of this specification, although there are
+>specific forms and instructions whose purpose is to aid in it.
+
+```
+interface-type ::= basictype
+  | structuretype 
+  | functionType
```

Comment:
nit: could you standardize the capitalization here to either all lowercaps or all camelCase?